docs: clarify rationales for oauth authentication (#3538)
kubeshop/testkube: docs/docs/guides/going-to-production/authentication/oauth-cli.md

# OAuth for CLI

> By default, testkube cli uses the "proxy" client which leverages the [kube apiserver proxy](https://kubernetes.io/docs/concepts/cluster-administration/proxies/) to reach the testkube api server. This implies granting users access to testkube internals in order to reach the testkube api REST endpoint (when using commands such as `testkube get artifact` )

Reviewer: ```suggestion
By default, Testkube CLI uses the "proxy" client which leverages the [kube apiserver proxy](https://kubernetes.io/docs/concepts/cluster-administration/proxies/) to reach the Testkube API server. This implies granting users access to Iestkube internals in order to reach the Testkube APIREST endpoint (when using commands such as `testkube get artifact` ).
```

> This section describes how to protect the testkube api-server REST api endpoint with [oauth2 authentication authorization grant](https://oauth.net/2/grant-types/authorization-code/), for use by the testkube cli as an oauth 2 client. In this mode, kubetest users do not need to be granted "kubetest administrator roles".

Reviewer: ```suggestion
This section describes how to protect the Testkube api-server REST API endpoint with [oauth2 authentication authorization grant](https://oauth.net/2/grant-types/authorization-code/), for use by the Testkube CLI as an oauth 2 client. In this mode, Testkube users do not need to be granted "Testkube administrator roles".
```

## Create Github OAuth Application

> Currently, only github OAuth authentication is supported. It is not yet possible to configure kube api-server to authenticate testkube cli oauth2 against other Oauth2 IDPs. 

Reviewer: ```suggestion
Currently, only GitHub OAuth authentication is supported. It is not yet possible to configure kube api-server to authenticate Testkube CLI OAuth2 against other OAuth2 IDPs. 
```

> In [oauth terminology](https://www.rfc-editor.org/rfc/rfc6749#section-1.1): 

Reviewer: ```suggestion
In [OAuth terminology](https://www.rfc-editor.org/rfc/rfc6749#section-1.1): 
```

> - github is the *authorization server*.

Reviewer: ```suggestion
- GitHub is the *authorization server*.
```

> - testkube cli is the *client* receiving http redirects from authorization server on a local http endpoint (http://127.0.0.1:13254) served by the cli. A local web browser invoked by [xdg-open](https://linux.die.net/man/1/xdg-open) is required to access github web ui, and then follow http redirect to the local http authorization callback endpoint.

Reviewer: ```suggestion
- Testkube CLI is the *client* receiving HTTP redirects from the authorization server on a local HTTP endpoint (http://127.0.0.1:13254) served by the CLI. A local web browser invoked by [xdg-open](https://linux.die.net/man/1/xdg-open) is required to access GitHub web UI, and then follow HTTP redirect to the local HTTP authorization callback endpoint.
```
